aten/src/ATen/native/Distance.cpp

```diff
@@ -25,6 +26,33 @@ Tensor pdist(const Tensor& self, const double p) {
   return at::_pdist_forward(self.contiguous(), p);
 }
 
+Tensor cdist(const Tensor& x1, const Tensor& x2, const double p) {
+    AT_CHECK(x1.dim() == 2, "cdist only supports 2D tensors, X1 got: ", x1.dim(), "D");
+    AT_CHECK(x1.is_contiguous(), "cdist requires X1 to be contiguous");
+    auto device1 = x1.type().device_type();
+    AT_CHECK(device1 == kCPU || device1 == kCUDA, "cdist only supports CPU and CUDA devices, X1 got: ", device);
+    AT_CHECK(x2.dim() == 2, "cdist only supports 2D tensors, X2 got: ", x2.dim(), "D");
+    AT_CHECK(x2.is_contiguous(), "cdist requires X2 to be contiguous");
+    auto device2 = x2.type().device_type();
+    AT_CHECK(device2 == kCPU || device2 == kCUDA, "cdist only supports CPU and CUDA devices, X2 got: ", device);
+    AT_CHECK(p >= 0, "cdist only supports non-negative p values");
+    int64_t c1 = x1.size(-1);
+    int64_t c2 = x2.size(-1);
+    AT_CHECK(c1 == c2, "X1 and X2 must have the same number of columns. X1: ", c1, " X2: ", c2);
+
+    int64_t r1 = x1.size(-2);
+    int64_t r2 = x2.size(-2);
+    Tensor result = at::empty({r1, r2}, x1.options());
+    if (r1 > 0) {
+        if (c1 == 0) {
+            result.fill_(0);
```

Comment:
why just this special case?  Any c1 == 0 is empty, as well as r2 == 0, right?

Reply:
Fixed